WordPress/gutenberg
# Add global inserter to Nav editor screen

## Description

Adds the global inserter to the Navigation block so as to achieve parity with other editors and also to provide users with an alternative means of adding new items.

**Question**: do we want/need e2e tests?

Addresses part of https://github.com/WordPress/gutenberg/issues/34268

Closes https://github.com/WordPress/gutenberg/issues/34438

## How has this been tested?




> What I found jarring is that unlike any other block the navigation links automatically open that LinkUI which is totally unexpected when adding from the inserter (cc @javierarce)

Note that this problem is not a facet of this PR. Rather it's a general problem for the Nav Editor. It should be solved in another PR.

To test:

* Enable experiment and go to new Navigation screen - you should see the global inserter in the top toolbar. It may be disabled.
* Create an empty menu - global inserter should be disable until you choose an option from the placeholder.
* Click `Start blank`.
* Use the global inserter to insert some blocks - it should close each time a block is inserted. **Question**: we need this behaviour on smaller screens where the panel obscures the main content area _but_ do we want this on larger screens?
* Try adding nested blocks - check that using the global inserter adds blocks in the expected place.
* Try selecting the top nav block - check that using the global inserter adds blocks after the last nav item on the top level.
* Try to select nothing (not even the Nav block) - check that the global inserter is not disabled and using the global inserter adds blocks after the last nav item on the top level.
* Add the following code to create a Custom Post type - this is required to trigger the `Browse All` below.
```php
function myplugin_register_book_post_type() {
	$args = array(
		'public'       => true,
		'label'        => 'Books',
		'show_in_rest' => true,
	);
	register_post_type( 'book', $args );
}
add_action( 'init', 'myplugin_register_book_post_type' );
```
* Use the Quick Inserter and click `Browser All`. 
    - Check this triggers the Global Inserter. 
    - Check that when you choose a block it is inserted at the location you were on when you triggered via the Quick Inserter.
* Check the mobile experience by repeating the above.

## Screenshots 


https://user-images.githubusercontent.com/444434/132684888-3b814f6a-0e32-4c6f-aab4-e1498ee42377.mov



## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
